src/cdef.rs

```diff
@@ -41,29 +43,37 @@ wrap_fn_ptr!(pub unsafe extern "C" fn cdef(
     dst_ptr: *mut DynPixel,
     stride: ptrdiff_t,
     left: *const [LeftPixelRow2px<DynPixel>; 8],
-    top: *const DynPixel,
-    bottom: *const DynPixel,
+    top_ptr: *const DynPixel,
+    bottom_ptr: *const DynPixel,
     pri_strength: c_int,
     sec_strength: c_int,
     dir: c_int,
     damping: c_int,
     edges: CdefEdgeFlags,
     bitdepth_max: c_int,
     _dst: *const FFISafe<Rav1dPictureDataComponentOffset>,
+    _top: *const FFISafe<DisjointMut<AlignedVec<u8, Align64<[u8; 64]>>>>,
+    _bottom: *const FFISafe<CdefBottom<'_>>,
 ) -> ());
 
+#[derive(Clone, Copy)]
+pub enum CdefBottom<'a> {
+    Pic(Rav1dPictureDataComponentOffset<'a>),
+    LineBuf((&'a DisjointMut<AlignedVec<u8, Align64<[u8; 64]>>>, usize)),
+}
+
 impl cdef::Fn {
     /// CDEF operates entirely on pre-filter data.
     /// If bottom/right edges are present (according to `edges`),
     /// then the pre-filter data is located in `dst`.
     /// However, the edge pixels above `dst` may be post-filter,
     /// so in order to get access to pre-filter top pixels, use `top`.
-    pub unsafe fn call<BD: BitDepth>(
+    pub fn call<BD: BitDepth>(
         &self,
         dst: Rav1dPictureDataComponentOffset,
         left: &[LeftPixelRow2px<BD::Pixel>; 8],
-        top: *const BD::Pixel,
-        bottom: *const BD::Pixel,
+        (top, top_off): (&DisjointMut<AlignedVec<u8, Align64<[u8; 64]>>>, usize),
```

Comment:
Can these be separate args?  No real reason to combine them into a tuple, right?

Reply:
Sure, done.